unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -0,0 +1,237 @@
+          expect(previewContainer.style.width).toEqual(`${width}px`);
+          expect(previewContainer.style.maxWidth).toEqual(`${width}px`);
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it('suggestionWidth change the width of the suggestion container', done => {
+        const suggestionWidth = '250px';
+        setupQuerySuggestPreview({ suggestionWidth });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        setTimeout(() => {
+          const suggestionContainerById = $$(suggestionContainer.el).find('#coveo-magicbox-suggestions');
+          expect(suggestionContainerById.style.minWidth).toBe(suggestionWidth);
+          expect(suggestionContainerById.style.maxWidth).toBe(suggestionWidth);
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it('headerText change the text in the header of the preview', done => {
+        const headerText = 'Super Header';
+        const suggestion = 'test';
+        setupQuerySuggestPreview({ headerText });
+        setupSuggestion();
+        triggerQuerySuggestHover(suggestion);
+        setTimeout(() => {
+          const previewContainer = $$(suggestionContainer.el).find('.coveo-preview-header > span');
+          expect(previewContainer.innerText).toBe(`${headerText} "${suggestion}"`);
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+    });
+
+    describe('when the previews are rendered,', () => {
+      it(`if we have one element,
+      it take 100% of the remaining available space`, done => {
+        setupQuerySuggestPreview({ numberOfPreviewResults: 1 });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        setTimeout(() => {
+          const previewContainer = $$(suggestionContainer.el).find('.CoveoResult');
+          expect(previewContainer.style.flex).toBe('0 0 100%');
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it(`if we have two element,
+      each take 50% of the remaining available space`, done => {
+        setupQuerySuggestPreview({ numberOfPreviewResults: 2 });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        setTimeout(() => {
+          const previewContainer = $$(suggestionContainer.el).find('.CoveoResult');
+          expect(previewContainer.style.flex).toBe('0 0 50%');
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it(`if we have three element,
+      each take 33% of the remaining available space`, done => {
+        setupQuerySuggestPreview({ numberOfPreviewResults: 3 });
+        setupSuggestion();
+        triggerQuerySuggestHover();
+        setTimeout(() => {
+          const previewContainer = $$(suggestionContainer.el).find('.CoveoResult');
+          expect(previewContainer.style.flex).toBe('0 0 33%');
+          done();
+        }, test.cmp.options.hoverTime);
+      });
+
+      it(`if we have four element,
```

Comment:
After reading the tests, I better understand the tests. I guess the main "exception" is the case were we have 4 results. What if we used 33% in this case, showing 3 results on the first row, and 1 on the second?

Right now we have the following progression where the index is the number of shown results:

```
100%, 50%, 33%, 50%, 33%, 33%
```
wheras we could simplify to:
```
100%, 50%, 33%, 33%, 33%, 33%
```

Alternatively, would it look bad if we just used 33% always? From a UI/UX view, wouldn't it be more consistent to always show the image at the same size? or is there too much white-space and it does not look good?

Reply:
Yeah I think you are right for when we have 1 or 2 elements, I will give it 33%

but for 4 elements, I asked Anne Catherine when she did the design how we should display we have 4 elements and she wanted to have a 2x2 grid instead of the 3x1